assignment.js

```diff
@@ -597,3 +597,89 @@ export function numberTransformer() {
     compose,
   };
 }
+/**
+ * This module provides functions to generate and process alphabets.
+ */
+export function alphabetProcessor() {
+  /**
+   * Generates an array of lowercase alphabets (a-z).
+   * @returns {string[]} Array of lowercase alphabets.
+   */
+  const generateAlphabets = () => {
+    const alphabetArray = [];
+    for (let i = 97; i <= 122; i++) {
+      alphabetArray.push(String.fromCharCode(i));
+    }
+    return alphabetArray;
+  };
+  /**
+   * Categorizes the given array of alphabets into vowels and consonants.
+   * @param {string[]} arrayOfAlphabets Array of lowercase alphabets.
+   * @returns {{ vowels: string[], consonants: string[] }} Object containing arrays of vowels and consonants.
+   */
+  const categorizeAlphabets = (arrayOfAlphabets) => ({
+    vowels: arrayOfAlphabets.reduce(
+      (acc, alphabet) =>
+        "aeiou".includes(alphabet) ? [...acc, alphabet] : acc,
+      []
+    ),
+    consonents: arrayOfAlphabets.reduce(
+      (acc, alphabet) =>
+        !"aeiou".includes(alphabet) ? [...acc, alphabet] : acc,
+      []
+    ),
+  });
+  /**
+   * Composes multiple functions into a single function.
+   * @param {...Function} functions - Functions to compose.
+   * @returns {Function} A composed function.
+   */
+  const compose =
+    (...functions) =>
+    (arguements) =>
+      functions.reduceRight((acc, func) => func(acc), arguements);
+  return {
+    generateAlphabets,
+    categorizeAlphabets,
+    compose,
+  };
+}
+/**
+ * A module for string processing utility functions.
+ * @module stringProcessor
+ */
+export function stringProcessor() {
+  /**
+   * Removes leading whitespace characters from a string.
+   * @param {string} str - The input string.
+   * @returns {string} - The string with leading whitespace characters removed.
+   */
+  const trimLeading = (str) => str.replace(/^\s+/g, "");
+  /**
+   * Removes trailing whitespace characters from a string.
+   * @param {string} str - The input string.
+   * @returns {string} - The string with trailing whitespace characters removed.
+   */
+  const trimTrailing = (str) => str.replace(/\s+$/g, "");
+  /**
+   * Reduces multiple consecutive whitespace characters to a single space.
+   * @param {string} str - The input string.
+   * @returns {string} - The string with consecutive whitespace characters reduced to a single space.
+   */
+  const singleSpace = (str) => str.replace(/\s+/g, " ");
+  /**
+   * Composes multiple functions into a single function.
+   * @param {...Function} functions - Functions to compose.
+   * @returns {Function} A composed function.
+   */
+  const compose =
+    (...functions) =>
+    (arguements) =>
+      functions.reduceRight((acc, func) => func(acc), arguements);
+  return {
+    trimLeading,
+    trimTrailing,
+    singleSpace,
+    compose,

```

Comment:
look we already duplicating compose, because of unwanted layers

Reply:
Implemented the changes